addons/binding/org.openhab.binding.dwdunwetter.test/src/test/java/org/openhab/binding/dwdunwetter/internal/data/DwdWarningsDataTest.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Test
+    public void testGetCeiling() {
+        assertThat(warningsData.getCeiling(0).format("%.0f ft"), is("9843 ft"));
+        assertThat(warningsData.getCeiling(1).format("%.0f ft"), is("9843 ft"));
+        assertThat(warningsData.getCeiling(2), is(UnDefType.NULL));
+    }
+
+    @Test
+    public void testGetExpires() {
+        // Conversion is needed as getExpires returns the Date with the System Default Zone
+        assertThat(((DateTimeType) warningsData.getExpires(0)).getZonedDateTime().withZoneSameInstant(ZoneId.of("UTC"))
+                .toString(), is("2018-12-22T18:00Z[UTC]"));
+        assertThat(((DateTimeType) warningsData.getExpires(1)).getZonedDateTime().withZoneSameInstant(ZoneId.of("UTC"))
+                .toString(), is("2018-12-23T01:00Z[UTC]"));
+        assertThat(warningsData.getExpires(2), is(UnDefType.NULL));
+    }
+
+    @Test
+    public void testGetOnset() {
+        // Conversion is needed as getOnset returns the Date with the System Default Zone
+        assertThat(((DateTimeType) warningsData.getOnset(0)).getZonedDateTime().withZoneSameInstant(ZoneId.of("UTC"))
+                .toString(), is("2018-12-21T10:00Z[UTC]"));
+        assertThat(((DateTimeType) warningsData.getOnset(1)).getZonedDateTime().withZoneSameInstant(ZoneId.of("UTC"))
+                .toString(), is("2018-12-22T18:00Z[UTC]"));
+        assertThat(warningsData.getOnset(2), is(UnDefType.NULL));
+    }
+
+    @Test
+    public void testGetEffective() {
+        // Conversion is needed as getEffective returns the Date with the System Default Zone
+        assertThat(((DateTimeType) warningsData.getEffective(0)).getZonedDateTime()
+                .withZoneSameInstant(ZoneId.of("UTC")).toString(), is("2018-12-22T03:02Z[UTC]"));
+        assertThat(((DateTimeType) warningsData.getEffective(1)).getZonedDateTime()
+                .withZoneSameInstant(ZoneId.of("UTC")).toString(), is("2018-12-22T10:15Z[UTC]"));
+        assertThat(warningsData.getEffective(2), is(UnDefType.NULL));
+    }
+
+    @Test
+    public void testGetWarning() {
+        assertThat(warningsData.getWarning(0), is(OnOffType.ON));
+        assertThat(warningsData.getWarning(1), is(OnOffType.ON));
+        assertThat(warningsData.getWarning(2), is(OnOffType.OFF));
+    }
+
+    @Test
+    public void testisNew() {
+        assertThat(warningsData.isNew(0), is(true));
+        assertThat(warningsData.isNew(1), is(true));
+        assertThat(warningsData.isNew(2), is(false));
+        // No longer new
+        assertThat(warningsData.isNew(0), is(false));
+        assertThat(warningsData.isNew(1), is(false));
+        assertThat(warningsData.isNew(2), is(false));
+    }
+
+    private void loadXmlFromFile() throws IOException {
+        InputStream stream = getClass().getResourceAsStream("warnings.xml");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream, Charset.forName("UTF-8")));
```

Comment:
`StandardCharsets.UTF_8` removes the `UnsupportedCharsetException`

Reply:
Thx.